src/new-components/modal/modal.test.tsx

```diff
@@ -189,6 +189,86 @@ describe('ModalFooter', () => {
     })
 })
 
+describe('ModalActions', () => {
+    it('renders a semantic contentinfo', () => {
+        render(<ModalActions data-testid="modal-actions">Hello</ModalActions>)
+        expect(screen.getByRole('contentinfo')).toBe(screen.getByTestId('modal-actions'))
+    })
+
+    it('does not acknowledge the className prop, but exceptionallySetClassName instead', () => {
+        render(
+            <ModalActions
+                data-testid="modal-actions"
+                // @ts-expect-error
+                className="wrong"
+                exceptionallySetClassName="right"
+            >
+                Hello
+            </ModalActions>,
+        )
+        const modalFooter = screen.getByTestId('modal-actions')
+        expect(modalFooter).toHaveClass('right')
+        expect(modalFooter).not.toHaveClass('wrong')
+    })
+
+    it('renders its children inlined inside it', () => {
+        render(
+            <ModalActions data-testid="modal-actions">
+                <button>OK</button>
+                <button>Cancel</button>
+            </ModalActions>,
+        )
+        expect(screen.getByTestId('modal-actions')).toMatchInlineSnapshot(`
+            <footer
+              class="box paddingTop-large paddingRight-large paddingBottom-large paddingLeft-large"
+              data-testid="modal-actions"
+            >
+              <div
+                class="space-large box display-flex flexDirection-row flexWrap-wrap alignItems-center justifyContent-flexEnd"
+              >
+                <button>
+                  OK
+                </button>
+                <button>
+                  Cancel
+                </button>
+              </div>
+            </footer>
+        `)
+    })
+
+    it('optionally renders a divider', () => {
+        const { rerender } = render(<ModalActions>Hello</ModalActions>)
+        expect(screen.queryByRole('separator')).not.toBeInTheDocument()
+        rerender(<ModalActions withDivider>Hello</ModalActions>)
+        expect(screen.getByRole('separator')).toBeInTheDocument()
+    })
+
+    it('ignores null children', () => {
+        render(
+            <ModalActions data-testid="modal-actions">
+                <button>OK</button>
+                <button>Cancel</button>
+                {null}
```

Comment:
Wouldn't this make it harder to debug `null` values as a consumer? Say we are passing down a `div` containing a `null` value. We would at least be able to see that the `div` is correctly being rendered, it just doesn't have a value. Not against it though, just wanted more regarding the decision to ignore them

Reply:
> Say we are passing down a `div` containing a `null` value.

What do you mean with "a `div` containing a `null` value"? And what do you mean with "debug `null` values as a consumer"?

Reply:
What I understand from the above is that if we have:

```
<ModalActions data-testid="modal-actions">
                <button>OK</button>
                <button>Cancel</button>
                <button>{some null value</button>
</ModalActions>
```

We would see:

```
<ModalActions data-testid="modal-actions">
                <button>OK</button>
                <button>Cancel</button>
 </ModalActions>
```

Is that right? Or would we render an empty button tag?

> And what do you mean with "debug null values as a consumer"?

If I've understood it right and the above example is how it would work, wouldn't it be confusing to eliminate the empty button tag with the `null` value? Especially as we inspect the DOM to see why it hasn't rendered?

Reply:
What the above example shows is that if one of the **direct** children of `ModalActions` is `null`, it will not generate a "slot" in the sequence of inlined elements that are spaced out. It has nothing to do with having non-null children that have their children `null`.

If this component that we're testing receives a `<button>{some null value}</button>`, it has no way of knowing that the component has null children. It only knows if its own children are null.